src/NumericalAlgorithms/Interpolation/Tags.hpp

```diff
@@ -40,8 +40,10 @@ struct InterpolationTargets {
 namespace Tags {
 
 /// Keeps track of which points have been filled with interpolated data.
+template <typename Metavariables>
```

Comment:
can we template these tags on the temporal_id::type instead of the Metavariables?

Reply:
Yes, I'm in the process of doing this.

In doing so, I see the following pattern multiple times.
What do you like better:

```
template<typename Metavariables>
void func(const Parallel::ConstGlobalCache<Metavariables>& cache,
          const typename Metavariables::temporal_id::type& temporal_id) noexcept {
      using TemporalId = typename Metavariables::temporal_id::type;
       ... code here that uses Metavariables and TemporalId...
}
```
or
```
template<typename Metavariables, typename TemporalId>
void func(const Parallel::ConstGlobalCache<Metavariables>& cache,
          const TemporalId& temporal_id) noexcept {
   ... code here that uses Metavariables and TemporalId...
}
```

The former is much longer and harder to read, but keeps the explicit relationship between
Metavariables and TemporalId.  I also have versions of the above with a `Requires<...something involving TemporalId...=nullptr>` in the template argument, which makes the former version even longer.

Reply:
looks with option 2), which is what @kidder and I prefer